bool connect_to_tool
bool disconect_from_tool
bool release_lf
bool grab_lf
